test/e2e/route_metrics_test.go

```diff
@@ -499,6 +549,38 @@ func TestRouteMetricsControllerRouteAndNamespaceSelector(t *testing.T) {
 		t.Fatalf("failed to fetch expected metrics: %v", err)
 	}
 
+	// Update the RouteSelector of the Ingress Controller so that the Route gets admitted again.
+	if err := kclient.Get(context.TODO(), routeFooLabelName, routeFooLabel); err != nil {
+		t.Fatalf("failed to get route resource: %v", err)
+	}
+	routeFooLabel.Labels = map[string]string{
+		"type": "rs-ns-bar",
+	}
+	if err := kclient.Update(context.TODO(), routeFooLabel); err != nil {
+		t.Fatalf("failed to update route: %v", err)
+	}
+
+	// Wait for metrics to be updated to 0 as the Route will get un-admitted by the IC.
+	if err := waitForRouteMetricsAddorUpdate(t, prometheusClient, ic.Name, 0); err != nil {
+		t.Fatalf("failed to fetch expected metrics: %v", err)
+	}
+
+	// Update the RouteSelector of the Ingress Controller so that the Route gets admitted again.
+	if err := kclient.Get(context.TODO(), routeFooLabelName, routeFooLabel); err != nil {
+		t.Fatalf("failed to get route resource: %v", err)
+	}
+	routeFooLabel.Labels = map[string]string{
+		"type": "rs-ns-foo",
+	}
+	if err := kclient.Update(context.TODO(), routeFooLabel); err != nil {
+		t.Fatalf("failed to update route: %v", err)
+	}
+
+	// Wait for metrics to be updated to 0 as the Route will get admitted by the IC again.
```

Comment:
```suggestion
	// Wait for metrics to be updated to 1 as the Route will get admitted by the IC again.
```

Reply:
Fixed the copy-paste issues. My bad :sweat_smile: